src/test/java/com/suse/saltstack/netapi/utils/ClientUtilsTest.java

```diff
@@ -64,4 +67,16 @@ public void testStreamToString() {
         assertEquals("Result doesn't match test string", TEST_STRING, result);
     }
 
+
+    @Test
+    public void testJoin() {
+        List<String> strings = Arrays.asList("Hello", "World", "123");
+        List<String> single = Arrays.asList("single");
+        List<String> empty = new ArrayList<>();
+
```

Comment:
If someone passes around `null` values then he deserves that the exception blows up in his face. :bomb: